File: app/src/main/java/com/tosh/poolandroid/ViewModel/RegistrationViewModel.java
package com.tosh.poolandroid.ViewModel;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.MutableLiveData;

import com.tosh.poolandroid.Remote.AuthRetrofitClient;
import com.tosh.poolandroid.Remote.NodeAuthService;

import com.tosh.poolandroid.RoomDb.UserRoom;

import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.disposables.CompositeDisposable;
import io.reactivex.functions.Consumer;
import io.reactivex.schedulers.Schedulers;

public class RegistrationViewModel  extends AndroidViewModel {
    private static MutableLiveData<String> registerResult = new MutableLiveData<>();

    private NodeAuthService api;
    private CompositeDisposable compositeDisposable = new CompositeDisposable();


    public RegistrationViewModel(@NonNull Application application) {
        super(application);
        api = AuthRetrofitClient.getInstance().create(NodeAuthService.class);

    }


    public void registerUser(final String name, final String email, String password, String confirmPassword) {
        compositeDisposable.add(api.registerUser(name,email,password,confirmPassword)
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(new Consumer<String>() {
                    @Override
                    public void accept(String res) throws Exception {

                        if(res.equals("successful")){
                            registerResult.postValue("success");
                        }else{
                           registerResult.postValue(res);
                        }
                    }
                }));
    }

    public MutableLiveData<String> getRegistrationResult(){
        return registerResult;
    }

    @Override
    protected void onCleared() {
        super.onCleared();
        compositeDisposable.clear();
    }
}

File: app/src/main/java/com/tosh/poolandroid/viewmodel/RegistrationViewModel.kt
package com.tosh.poolandroid.viewmodel

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import com.tosh.poolandroid.model.RegisterResponse
import com.tosh.poolandroid.Remote.RetrofitClient
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class RegistrationViewModel:ViewModel(){
    fun userRegister(name:String, email:String, password:String, confirmPassword:String):LiveData<String>{
        val registerResponse = MutableLiveData<String>()
        RetrofitClient.makeRetrofitApi().userRegister(name, email, password, confirmPassword)
            .enqueue(object : Callback<RegisterResponse> {
                override fun onFailure(call: Call<RegisterResponse>, t: Throwable) {
                    registerResponse.value = t.message
                }

                override fun onResponse(call: Call<RegisterResponse>, response: Response<RegisterResponse>) {
                    if (response.isSuccessful){
                        registerResponse.value = response.body()?.message.toString()
                    }else{
                        registerResponse.value = response.body()?.message.toString()
                    }
                }

            })
        return registerResponse
    }

}